RadarSDK/RadarNotificationHelper.m

```diff
@@ -68,4 +74,149 @@ + (void)showNotificationsForEvents:(NSArray<RadarEvent *> *)events {
+                [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelInfo message:[NSString stringWithFormat:@"Found pending notification | identifier = %@", request.identifier]];
+                [identifiers addObject:request.identifier];
+            }
+        }
+
+        if (identifiers.count > 0) {
+            [notificationCenter removePendingNotificationRequestsWithIdentifiers:identifiers];
+            [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelDebug message:@"Removed pending notifications"];
+        }
+
+        completionHandler();
+    }];
+}
+
++ (void)addOnPremiseNotificationRequests:(NSArray<UNNotificationRequest *> *)requests {
+    UNUserNotificationCenter *notificationCenter = [UNUserNotificationCenter currentNotificationCenter];
+    [notificationCenter getNotificationSettingsWithCompletionHandler:^(UNNotificationSettings *settings) {
+        if (settings.authorizationStatus != UNAuthorizationStatusAuthorized) {
+            [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelInfo message:@"Notification permissions not granted. Skipping adding notifications."];
+            return;
+        }
+        
+        for (UNNotificationRequest *request in requests) {
+            [notificationCenter addNotificationRequest:request withCompletionHandler:^(NSError *_Nullable error) {
+                if (error) {
+                    [[RadarLogger sharedInstance]
+                        logWithLevel:RadarLogLevelInfo
+                            message:[NSString stringWithFormat:@"Error adding local notification | identifier = %@; error = %@", request.identifier, error]];
+                } else {
+                    [RadarState addPendingNotificationRequest:request];
+                    [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelInfo
+                                                       message:[NSString stringWithFormat:@"Added local notification | identifier = %@", request.identifier]];
+                }
+            }];
+        }
+    }];
+}
+
++ (void)registerBackgroundNotificationChecks {
+    if (@available(iOS 13.0, *)) {
+        [[BGTaskScheduler sharedScheduler] registerForTaskWithIdentifier:@"io.radar.notificationCheck" usingQueue:nil launchHandler:^(BGTask *task) {
+            [self handleAppRefreshTask:task];
+        }];
+    }
+}
+
++ (void)scheduleBackgroundNotificationChecks {
+    if (@available(iOS 13.0, *)) {
+        BGAppRefreshTaskRequest *request = [[BGAppRefreshTaskRequest alloc] initWithIdentifier:@"io.radar.notificationCheck"];
+        request.earliestBeginDate = [NSDate dateWithTimeIntervalSinceNow:60*60];
+        NSError *error = nil;
+        
+        [[BGTaskScheduler sharedScheduler] submitTaskRequest:request error:&error];
+        if (error) {
+            [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelError message:[NSString stringWithFormat:@"Error scheduling app refresh task: %@", error]];
+        }
```

Comment:
Seeing the following in Waypoint:
```
Error scheduling app refresh task: Error Domain=BGTaskSchedulerErrorDomain Code=3 "(null)"
```

Reply:
We get this when we try and schedule on the simulator.